File: app/src/main/java/com/yuchengren/demo/util/business/MenuUtil.java
package com.yuchengren.demo.util.business;

import android.content.Context;
import android.content.Intent;

import com.yuchengren.demo.app.other.CallPhoneBackActivity;
import com.yuchengren.demo.app.other.ImageEditActivity;
import com.yuchengren.demo.app.other.GridActivity;
import com.yuchengren.demo.app.other.RxAndroidActivity;
import com.yuchengren.demo.app.test.TestActivity;
import com.yuchengren.demo.app.other.ChangeThemeActivity;
import com.yuchengren.demo.constant.MapKey;
import com.yuchengren.demo.constant.MenuCode;
import com.yuchengren.demo.entity.db.MenuEntity;

/**
 * Created by yuchengren on 2017/12/20.
 */

public class MenuUtil {

	public static void startActivity(Context context,MenuEntity menuEntity){
		if(menuEntity == null || menuEntity.getCode() == null){
			return;
		}
		Intent intent;
		switch (menuEntity.getCode()){
			case MenuCode.Second.CALL_PHONE_BACK:
				intent = new Intent(context, CallPhoneBackActivity.class);
				break;
			case MenuCode.Second.RX_ANDROID:
				intent = new Intent(context, RxAndroidActivity.class);
				break;
			case MenuCode.Second.TEST:
				intent = new Intent(context, TestActivity.class);
				break;
			case MenuCode.Second.THEME_SWITCH:
				intent = new Intent(context, ChangeThemeActivity.class);
				break;
			case MenuCode.Second.IMAGE_EDIT:
				intent = new Intent(context, ImageEditActivity.class);
				break;
			case MenuCode.Second.NINE_GRID:
				intent = new Intent(context, GridActivity.class);
				break;
			default:
				intent = new Intent();
				break;
		}
		intent.putExtra(MapKey.CODE,menuEntity.getCode());
		intent.putExtra(MapKey.NAME,menuEntity.getName());
		context.startActivity(intent);
	}
}

File: app/src/main/java/com/yuchengren/demo/MenuUtil.kt
package com.yuchengren.demo

import android.content.Context
import android.content.Intent

import com.yuchengren.demo.app.adaptation.AndroidEightAdaptationActivity
import com.yuchengren.demo.app.adaptation.AndroidNineAdaptationActivity
import com.yuchengren.demo.app.adaptation.AndroidSevenAdaptationActivity
import com.yuchengren.demo.app.other.ChangeThemeActivity
import com.yuchengren.demo.app.other.CallPhoneBackActivity
import com.yuchengren.demo.app.other.ImageEditActivity
import com.yuchengren.demo.app.other.GridActivity
import com.yuchengren.demo.app.other.RxAndroidActivity
import com.ycr.module.photo.chosen.ChosenPhotoActivity
import com.ycr.module.base.constant.MapKey
import com.ycr.module.base.constant.MenuCode
import com.ycr.module.base.entity.db.MenuEntity
import com.yuchengren.demo.app.body.main.MainActivity
import com.yuchengren.demo.app.test.KotlinTestActivity
import com.yuchengren.demo.app.test.TestActivity
import com.yuchengren.demo.app.widgettest.refreshview.RefreshViewActivity

/**
 * Created by yuchengren on 2017/12/20.
 */

object MenuUtil {

    @JvmStatic fun startActivity(context: Context, menuEntity: MenuEntity?) {
        if (menuEntity == null || menuEntity.code == null) {
            return
        }
        val clazz = when (menuEntity.code) {
            MenuCode.Second.CALL_PHONE_BACK -> CallPhoneBackActivity::class.java
            MenuCode.Second.RX_ANDROID -> RxAndroidActivity::class.java
            MenuCode.Second.TEST -> TestActivity::class.java
            MenuCode.Second.KOTLIN_TEST -> KotlinTestActivity::class.java
            MenuCode.Second.THEME_SWITCH -> ChangeThemeActivity::class.java
            MenuCode.Second.IMAGE_EDIT -> ImageEditActivity::class.java
            MenuCode.Second.NINE_GRID -> GridActivity::class.java
            MenuCode.Second.CHOOSE_PHOTO -> ChosenPhotoActivity::class.java
            MenuCode.Second.SEVEN_ADAPTATION -> AndroidSevenAdaptationActivity::class.java
            MenuCode.Second.EIGHT_ADAPTATION -> AndroidEightAdaptationActivity::class.java
            MenuCode.Second.NINE_ADAPTATION -> AndroidNineAdaptationActivity::class.java
            MenuCode.Second.REFRESH_VIEW -> RefreshViewActivity::class.java
            else -> MainActivity::class.java
        }
        val intent: Intent = Intent(context,clazz).apply {
            putExtra(MapKey.CODE, menuEntity.code)
            putExtra(MapKey.NAME, menuEntity.name)
        }
        context.startActivity(intent)
    }
}
